### PR TITLE
logcheck: pass all packages at once to logcheck

### PR DESCRIPTION
**Pull request by umangachapagain**
Signed-off-by: Umanga Chapagain <chapagainumanga@gmail.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
passing all packages at once to logcheck reduces the time
taken by linter.

#### Which issue(s) this PR fixes:

Ref kubernetes/klog/issues/223

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: hack/verify-structured-logging.sh [hunk 1]
```diff
@@ -49,12 +49,9 @@ while IFS='' read -r line; do
 done < <(cat "${migrated_packages_file}")
 
 
-# TODO: Improve concurrancy here
 ret=0
-for package in "${migrated_packages[@]}"; do
-    GOOS=linux    logcheck "$KUBE_ROOT/$package" || ret=$?
-    GOOS=windows  logcheck "$KUBE_ROOT/$package" || ret=$?
-done
+GOOS=linux    logcheck "$KUBE_ROOT/$migrated_packages" || ret=$?
```
Comment (adisky) [inline hack/verify-structured-logging.sh:53]: you need to expand the array here. Also shell check failure for the same 
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/100246/pull-kubernetes-verify/1371430538098774016#1:build-log.txt%3A2129

Comment (umangachapagain) [reply hack/verify-structured-logging.sh]: Fixed it. That explains why lint finished that fast.

**Comment by adisky** [comment]
Thanks @umangachapagain for working on this!!!

It is significantly reducing the time, I have checked randomly on 2-3 PR's it is around ~ 115 seconds. With this it is around 
67s. Also some research done by @umangachapagain [here](https://github.com/kubernetes/klog/issues/223) to speed up the linter 

I have also verified running locally including some failing packages.

/sig testing instrumentation
/cc @BenTheElder @ehashman 

Ref:
on this pr: 67s
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/100246/pull-kubernetes-verify/1371476671391076352#1:build-log.txt%3A2160

on other pr's: > 110s
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/100279/pull-kubernetes-verify/1371742218225717248#1:build-log.txt%3A2159

**Comment by BenTheElder** [comment]
excellent, thank you!

**Comment by BenTheElder** [comment]
milestone-ing test-only change prior to test-freeze

**Pull request merged**